picoquictest/stresstest.c

```diff
@@ -0,0 +1,875 @@
+        DBG_PRINTF("Cannot create the client context #%d.\n", (int)client_index);
+        ret = -1;
+    }
+    if (ret == 0) {
+        memset(ctx, 0, sizeof(picoquic_stress_client_t));
+        /* Initialize client specific address */
+        stress_set_ip_address_from_index(&ctx->client_addr, (int)client_index);
+        /* set stream ID to default value */
+
+        /* initialize client specific ticket file */
+        memcpy(ctx->ticket_file_name, "stress_ticket_000.bin", 19);
+        ctx->ticket_file_name[14] = (uint8_t)('0' + client_index / 100);
+        ctx->ticket_file_name[15] = (uint8_t)('0' + (client_index / 10) % 10);
+        ctx->ticket_file_name[16] = (uint8_t)('0' + client_index % 10);
+        ctx->ticket_file_name[21] = 0;
+        if (ret == 0) {
+            ret = picoquic_save_tickets(NULL, stress_ctx->simulated_time, ctx->ticket_file_name);
+            if (ret != 0) {
+                DBG_PRINTF("Cannot create ticket file <%s>.\n", ctx->ticket_file_name);
+            }
+        }
+    }
+    if (ret == 0) {
+        /* initialize the simulation links from client to server and back. */
+        ctx->c_to_s_link = picoquictest_sim_link_create(0.01, 10000, 0, 0, 0);
+        ctx->s_to_c_link = picoquictest_sim_link_create(0.01, 10000, 0, 0, 0);
+        if (ctx->c_to_s_link == NULL ||
+            ctx->s_to_c_link == NULL) {
+            DBG_PRINTF("Cannot create the sim links for client #%d.\n", (int)client_index);
+            ret = -1;
+        }
+    }
+
+    if (ret == 0) {
+        /* Create the quic context for this client*/
+        ctx->qclient = picoquic_create(8, NULL, NULL, NULL, NULL,
+            NULL, NULL, NULL, NULL, stress_ctx->simulated_time, &stress_ctx->simulated_time,
+            ctx->ticket_file_name, NULL, 0);
+        if (ctx->qclient == NULL) {
+            DBG_PRINTF("Cannot create the quic client #%d.\n", (int)client_index);
+            ret = -1;
+        }
+    }
+
+    return ret;
+}
+
+static void stress_delete_client_context(int client_index, picoquic_stress_ctx_t * stress_ctx)
+{
+    picoquic_stress_client_t * ctx = stress_ctx->c_ctx[client_index];
+
+    if (ctx != NULL) {
+        if (ctx->qclient != NULL) {
+            picoquic_free(ctx->qclient);
+            ctx->qclient = NULL;
+        }
+
+        if (ctx->c_to_s_link != NULL) {
+            picoquictest_sim_link_delete(ctx->c_to_s_link);
+            ctx->c_to_s_link = NULL;
+        }
+
+        if (ctx->s_to_c_link != NULL) {
+            picoquictest_sim_link_delete(ctx->s_to_c_link);
+            ctx->s_to_c_link = NULL;
+        }
+        free(ctx);
+
+        stress_ctx->c_ctx[client_index] = NULL;
+    }
+}
+
+int stress_test()
+{
+    int ret = 0;
+    picoquic_stress_ctx_t stress_ctx;
+
+    /* Initialization */
+    memset(&stress_ctx, 0, sizeof(picoquic_stress_ctx_t));
+    stress_set_ip_address_from_index(&stress_ctx.server_addr, -1);
+    if (stress_ctx.nb_clients > PICOQUIC_MAX_STRESS_CLIENTS) {
```

Comment:
The context is initialized to `0` two lines above, so this check is not required.

Reply:
Or initialize the field before testing it... Thanks for finding that.